contrib/epee/include/net/levin_protocol_handler_async.h

```diff
@@ -953,12 +947,14 @@ int async_protocol_handler_config<t_connection_context>::send(byte_slice message
 template<class t_connection_context>
 bool async_protocol_handler_config<t_connection_context>::close(boost::uuids::uuid connection_id)
 {
-  CRITICAL_REGION_LOCAL(m_connects_lock);
-  async_protocol_handler<t_connection_context>* aph = find_connection(connection_id);
-  if (!aph)
+  async_protocol_handler<t_connection_context>* aph = nullptr;
+  if (find_and_lock_connection(connection_id, aph) != LEVIN_OK)
     return false;
+  auto scope_exit_handler = misc_utils::create_scope_leave_handler(
+    boost::bind(&async_protocol_handler<t_connection_context>::finish_outer_call, aph));
   if (!aph->close())
     return false;
+  CRITICAL_REGION_LOCAL(m_connects_lock);
   m_connects.erase(connection_id);
```

Comment:
This `erase` call cannot be outside of the lock, and therefore I don't see the benefits to the changes in this function.

Reply:
Sorry read through this code too quickly earlier, this is moving the lock call downward instead of doing a `CRITICAL_REGION_END` call.

This is now acquiring the lock twice. I'm not immediately seeing where deadlock would occur, after running through the call sequence of `aph->close()`. If I revert these changes, the unit test will deadlock here? This call is used in many places throughout the code, so I'm a little skeptical is necessary.